.ci/lint

```diff
@@ -30,40 +38,38 @@ pipeline {
                 python --version
                 pip install tox
                 '''
+                archiveArtifacts artifacts: 'file-list-status.log,file-list-changed.log,file-list-deleted.log'
             }
         }
         stage('linting') {
             failFast false
             parallel {
                 stage('salt linting') {
+                    when {
+                        expression { return readFile('file-list-changed.log') ==~ /(?i)setup.py|salt\/.*\.py/ }
+                    }
                     steps {
                         sh '''
                         eval "$(pyenv init - --no-rehash)"
-                        _FILES="$(find salt/ -name "*.py" -exec git diff --name-only "origin/$CHANGE_TARGET" "origin/$BRANCH_NAME" {} +)"
-                        _FILES="$_FILES $(git diff --name-only "origin/$CHANGE_TARGET" "origin/$BRANCH_NAME" setup.py)"
-                        if [[ -z ${_FILES} ]]; then
-                            echo "No pylint run, no changes found in the files"
-                            echo "empty" pylint-reports.xml
-                        else
-                            tox -e pylint-salt ${_FILES} | tee pylint-report.xml
-                        fi
+                        egrep -E '^salt/.*\\.py$|^setup\\.py$' file-list-changed.log | xargs -r '--delimiter=\\n' tox -e pylint-salt | tee pylint-report-salt.log
```

Comment:
We're still doing `egrep` here, please remove that first `e`, it should be just `grep -E`.